app/main.ipynb

```diff
@@ -112,15 +165,30 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "outputs": [],
+   "execution_count": 91,
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "0.18566408063125786"
+     },
+     "execution_count": 91,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
-    "# write your code for task 3 here"
+    "poland_count = len(data[data[\"native-country\"] == \"Poland\"])\n",
+    "percentage_from_poland = (poland_count / len(data)) * 100\n",
+    "percentage_from_poland"
```

Comment:
To further optimize the performance of your code, especially if dealing with large datasets, you might consider using pandas' built-in functions like value_counts which can be more efficient. This approach could enhance the speed of your analysis.